dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthenticator.java

```diff
@@ -46,10 +49,35 @@ public SimpleAuthenticator(TokenDAO dao, UserDAO userDAO) {
     @Override
     public Optional<User> authenticate(String credentials) throws AuthenticationException {
         LOG.debug("SimpleAuthenticator called with {}", credentials);
+        if (isGoogleToken(credentials)) {
+            final Optional<String> username = GoogleHelper.getUserNameFromToken(credentials);
+            if (username.isPresent()) {
+                User user = userDAO.findByUsername(username.get());
+                if (user != null) {
+                    List<Token> tokens = dao.findByUserId(user.getId());
+                    Token googleToken = Token.extractToken(tokens, TokenType.GOOGLE_COM);
+                    googleToken.setContent(credentials);
+                    dao.update(googleToken);
+                    return Optional.of(user);
+                }
+            }
+        }
+        // Just in case the check for whether the credentials are a Google token are incorrect,
+        // try it as a Dockstore token.
         final Token token = dao.findByContent(credentials);
         if (token != null) {
             return Optional.of(userDAO.findById(token.getUserId()));
         }
         return Optional.empty();
     }
+
+    /**
+     * A crude test for whether a token is a Dockstore token or a Google access token.
+     *
+     * @param credentials
+     * @return
+     */
+    static boolean isGoogleToken(String credentials) {
+        return credentials != null && credentials.startsWith("ya29.");
```

Comment:
Could use a comment about the prefix ya29.

Reply:
Good comment, which opened a can of worms. :) I decided to get rid of this method entirely, because it was a fragile check. I changed the `authenticate` method to instead assume it's a Dockstore token and look it up; if that fails, then we try to see if it's a Google token.